app.js

```diff
@@ -403,92 +403,102 @@ let DNSResult = create({
   displayName: "DNS",
 
   render: function() {
-    let j = this.props.json;
-    if (j.SRVRecords == null) {
-      return (
-        <div className="dns">
-          <h2>No SRV Records</h2>
-        </div>
-      );
-    }
+    const j = this.props.json;
 
-    let records = j.SRVRecords.map((record, id) => {
-      return (
-        <div className="row" key={id}>
-          <div className="col">{record.Target}</div>
-          <div className="col">{record.Port}</div>
-          <div className="col">{record.Priority}</div>
-          <div className="col">{record.Weight}</div>
-        </div>
-      );
-    });
+    const skippedSrvResultTitle = <p>server name/.well-known result contains explicit port number: no SRV lookup done</p>;
+    const foundSrvResultTitle = <h3>DNS records for {j.SRVCName}</h3>;
+    const srvResultTitle = j.SRVSkipped ? skippedSrvResultTitle : foundSrvResultTitle;
 
-    let hosts = Object.keys(j.Hosts).map((host) => {
-      if (j.Hosts[host].Addrs != null) {
-        return j.Hosts[host].Addrs.map((address, id) => {
+    const srvRecordsTable = function() {
+      if (j.SRVRecords != null) {
+        const recordRows = j.SRVRecords.map((record, id) => {
```

Comment:
I've been doing some testing on this, and generally it now looks great for servers with SRV records, and those where the SRV lookup is skipped.

However, there is a third case we're not correctly handling, which is when the SRV lookup is not skipped, but there are no SRV records - see for example the results for `sw1v.org`:

![image](https://user-images.githubusercontent.com/1389908/132004647-dcfb71ec-4ce2-48b6-8c35-20d546c55b21.png)

In this case, `SRVSkipped` is false, but `SRVCName` is `""` and `SRVRecords` is `null`.

How about moving the logic for the title into this function:

```suggestion
    const srvRecordsTable = function() {
      if (j.SRVSkipped) {
        return <p>server name/.well-known result contains explicit port number: no SRV lookup done</p>;
      } else if (j.SRVRecords == null) {
        return <p>No SRV records found</p>;
      } else { 
        const recordRows = j.SRVRecords.map((record, id) => {
```

(and add `<h3>DNS records for {j.SRVCName}</h3>` further down)

Then `srvResultTitle` and friends can be removed.

Reply:
This looks defensively weird. Thx for catching it :see_no_evil: